### DISCUSSION TITLE
Watchers inside setup functions don't get invoked

### DISCUSSION THREAD

**Discussion by lorefnon** [vuejs/pinia#1792] [Help and Questions] [answered]
### Reproduction

https://stackblitz.com/edit/github-3uy8yt?file=src/App.vue

### Steps to reproduce the bug

Looking at various [discussions](https://github.com/vuejs/pinia/discussions/794), it seems like I should be able to define a watcher inside a store setup function. However in practice it doesnt seem to work. 

In the linked stackblitz I have modified the official example to use setup functions, and have two watchers defined: 

First is inside cart.js: 

```js
export const useCartStore = defineStore('cart', () => {
  const rawItems = ref([]);

....

  watch(rawItems, (oldItems, newItems) => {
    console.log('Watcher invoked', oldItems, newItems);
  });

....
})
```

Second is inside App.vue: 

```js
    const cart = useCartStore();

    watch(cart.rawItems, (prevItems, newItems) => {
      console.log('component level watcher: ', prevItems, newItems);
    });

```

### Expected behavior

Both watchers get invoked whenever an item gets added/removeed to the cart

### Actual behavior

The second gets invoked everytime an item is added or removed. But the first never gets invoked.

**Comment by posva** [comment, marked as answer]
You need a deep watcher for arrays (https://vuejs.org/guide/essentials/watchers.html). You will notice the same happens [in plain vue](https://sfc.vuejs.org/#eNo9UMtuwzAM+xXBl6Zoam/XIC027DPmHdLMadLFD1hKewj875OdrTeLNCmKq3gPQd4XIxrRYh+nQICGlnDWbrLBR4IVohlqeHTUj5BgiN7CjhU77bTrvUMCi1c45W/Vy/4fm8hY/EM/vxjWrlhUhaih2sPpDKt2kOXy3s2LORy0SzVv/DYmNDB0MxpIrG3Vlo1T8cAGYe7I8ATQhvO6lgQptSrn1tReFiLv4K2fp/7npEXZKcOCY/W616IIAT4yC9YUG7VpmGrVc4GoxdbC0XZB3tA77qlkZs9CoBbNdkXGuJY8azESBWyUwqHP7d5Q+nhV/JJxcTRZIw3a4yX6B5rIxlpki8Tni/QLVyaJ9g==)

BTW the second watcher should be:

```ts
watch(() => cart.rawItems, ..., { deep: true })
```

There is a small section explaining this in the same doc page

**Comment by lorefnon** [comment]
Thank you. 

Why is the arrow function needed in the second example though ?

**Reply by posva** [reply]
See https://vuejs.org/guide/essentials/watchers.html#basic-example, a store is a reactive object
